### DISCUSSION TITLE
clarification on create() / best practices for multiple components invoking a store

### DISCUSSION THREAD

**Discussion by dsvoid** [pmndrs/zustand#1942] [Q&A]
My question is related to the performance of zustand when multiple components are accessing the same store.

- I have an `XStore` with state variables and handlers.
- I have written a `useXStore` function call that returns the output of a `create` on the store.
- My application renders a list of components which individually have a `const store = useXStore()`.

Should I be mindful to avoid making these multiple calls as the list of components grows larger? My naive assumption was first that each component calling `useXStore()` is in fact accessing a single store, and that the first call creates the store that every other component uses, but I don't actually know what's going on under the hood, and I would appreciate some clarification. Supposing this isn't the case, should I employ another method of accessing store? My off-the-cuff implementation ideas:

- move the `useXStore` call into the parent of the listed components and use callbacks in each child component to the parent to perform operations
- ensure that the initial result of the first `useXStore` call is saved as a global, and that subsequent calls to `useXStore` return this global instead of running `create` again

**Comment by samhirtarif** [comment]
@dsvoid - As long as you're doing something along these lines

```js
const useXStore = create((set) => ({
  someValue: 42,
  increaseValue: () => set((state) => ({ someValue: state.someValue + 1 })),
}))
```

And then using the `useXStore` in your code or in your components as below, it would only create one instance of the store.
```js
const someValue = useXStore().someValue;
```

This is something tried and tested.
